zone/client_packet.cpp

```diff
@@ -8916,12 +8916,30 @@ void Client::Handle_OP_ItemVerifyRequest(const EQApplicationPacket *app)
 			}
 			else if (item->ItemType == EQ::item::ItemTypeSpell)
 			{
-				if (RuleB(Spells, AllowSpellMemorizeFromItem)) {
+				spell_id = item->Scroll.Effect;
+				if (RuleB(Spells, AllowSpellMemorizeFromItem)) 
+				{
+					int highest_spell_id = GetHighestScribedSpellinSpellGroup(spells[spell_id].spell_group);
+					if (spells[spell_id].spell_group > 0 && highest_spell_id > 0) 
+					{
+						if (spells[spell_id].rank > spells[highest_spell_id].rank)
+						{
+							std::string message = fmt::format("{} will replace {} in your spellbook", GetSpellName(spell_id), GetSpellName(highest_spell_id));
+							this->SetEntityVariable("slot_id",itoa(slot_id));
+							this->SetEntityVariable("spell_id",itoa(item->ID));
+							this->SendPopupToClient("", message.c_str(), 1000001, 1, 10);
+							return;
+						} 
+						else if (spells[spell_id].rank < spells[highest_spell_id].rank)
+						{
+							MessageString(Chat::Red, 11004, spells[spell_id].name, spells[highest_spell_id].name);
+							return;
+						} 
+					}
 					DeleteItemInInventory(slot_id, 1, true);
 					MemorizeSpellFromItem(item->ID);
-				} else {
```

Comment:
Removing the else intentional here ?

Reply:
Yes, this is intentional.

Reply:
Added back in, since it logically makes sense to return if the rule isn't in use.